test/test_slots.py

```diff
@@ -66,6 +66,88 @@ def test_tp_destroy(self):
         gc.collect()
         assert mod.get_destroyed_x() == 7
 
+    def test_tp_finalize_nongc(self):
+        import gc
+        mod = self.make_module("""
+            @DEFINE_PointObject
+            @DEFINE_Point_new
+
+            static long finalized_x;
+
+            HPyDef_SLOT(Point_finalize, Point_finalize_impl, HPy_tp_finalize)
+            static void Point_finalize_impl(HPyContext *ctx, HPy obj)
+            {
+                PointObject *point = PointObject_AsStruct(ctx, obj);
+                finalized_x += point->x;
+            }
+
+            HPyDef_METH(get_finalized_x, "get_finalized_x", get_finalized_x_impl, HPyFunc_NOARGS)
+            static HPy get_finalized_x_impl(HPyContext *ctx, HPy self)
+            {
+                return HPyLong_FromLong(ctx, finalized_x);
+            }
+
+            @EXPORT_POINT_TYPE(&Point_new, &Point_finalize)
+            @EXPORT(get_finalized_x)
+            @INIT
+       """)
+        point = mod.Point(7, 3)
+        assert mod.get_finalized_x() == 0
+        del point
+        gc.collect()
+        assert mod.get_finalized_x() == 7
+        gc.collect()
```

Comment:
In Graalpython, our `gc.collect()` does not ensure that the GC is really finished after the call (could be a parallel GC) and also does not ensure that the GC will collect that object (if there is no memory pressure, the GC could decide to just leave it).

Long story short: Could you add something like `supports_sync_gc` to class `HPyTest` (just like `supports_refcounts`) and use it here?

Reply:
@fangerer Is there a way to ensure GC collection on GraalPython? Perhaps we can instead have a `.gc_collect()` helper method that PyPy and GraalPython could implement differently if needed? Otherwise it becomes very hard to test these kinds of things.

Reply:
No, sorry, I don't see a way to ensure that. What we usually do is to repeat the test in a loop (with a maximum number of retries) and also insert a sleep to give the GC enough time. But that's also just best-effort and does not ensure success.

What about introducing a custom pytest mark for such tests?

Reply:
Adding a pytest mark seems like a good solution - CPython and pypy would just ignore it and GraalPython would deal with it however it wants. I'll let the GraalPython folks implement it though, since you already know which tests cause issues and how to handle them.

Reply:
Okay, I'll implement it and push to this PR.